src/Model/Assertion.php

```diff
@@ -74,26 +73,20 @@ public function buildRoleLinks(RoleManager $rm): void
     {
         $this->rm = $rm;
         $count = substr_count($this->value, '_');
+        if($count < 2) {
+            throw new CasbinException('the number of "_" in role definition should be at least 2');
+        }
 
         foreach ($this->policy as $rule) {
-            if ($count < 2) {
-                throw new CasbinException('the number of "_" in role definition should be at least 2');
-            }
             if (\count($rule) < $count) {
                 throw new CasbinException('grouping policy elements do not meet role definition');
             }
-
-            if (2 == $count) {
-                $this->rm->addLink($rule[0], $rule[1]);
-            } elseif (3 == $count) {
-                $this->rm->addLink($rule[0], $rule[1], $rule[2]);
-            } elseif (4 == $count) {
-                $this->rm->addLink($rule[0], $rule[1], $rule[2], $rule[3]);
+            if(\count($rule) > $count) {
+                $rule = \array_slice($rule, 0, $count);
             }
-        }
 
-        Log::logPrint('Role links for: ' . $this->key);
-        $this->rm->printRoles();
```

Comment:
This shouldn't be removed.

Reply:
Really? This code has been removed from the Core Library.